openedx_tagging/core/tagging/rest_api/v1/permissions.py

```diff
@@ -0,0 +1,27 @@
+"""
+Taxonomy permissions
+"""
+
+from rest_framework.permissions import DjangoObjectPermissions
+
+
+class TaxonomyObjectPermissions(DjangoObjectPermissions):
+    perms_map = {
+        "GET": ["%(app_label)s.view_%(model_name)s"],
+        "OPTIONS": [],
+        "HEAD": ["%(app_label)s.view_%(model_name)s"],
+        "POST": ["%(app_label)s.add_%(model_name)s"],
+        "PUT": ["%(app_label)s.change_%(model_name)s"],
+        "PATCH": ["%(app_label)s.change_%(model_name)s"],
+        "DELETE": ["%(app_label)s.delete_%(model_name)s"],
+    }
+
+    def has_permission(self, request, view):
+        # Workaround to allow 'retrieve' view to pass through the
+        # method GET permission check. Actual object permission check
+        # is done in the get_object() method.
+        if view.action == "retrieve":
+            return bool(request.user and request.user.is_authenticated)
+        else:
+            return super().has_permission(request, view)
+
```

Comment:
I can see from Jira that you had a question about permissions, but I don't see it on the PR?

Basically, I'm suggesting that we don't need this method at all, and so the view permissions will use the rules we've established, namely that that [only `staff` (or `superuser`s) should be allowed to view taxonomies](https://github.com/open-craft/openedx-learning/blob/963841c1cf9625b0e6bace6f17e3475e9c1642c4/openedx_tagging/core/tagging/rules.py#L16-L22), or do anything else with them.

Normal users cannot even view them, but that's what this method allows, so it should be removed, and any failing tests updated accordingly.

```suggestion
```

Reply:
The rule (the one you linked) allows ANY user to view an `{enabled=true}` taxonomy. That's why I'm confused. If we want to ensure that only staff should be allowed to view taxonomies, then we should rewrite the rule as follows:

```python
@rules.predicate
def can_view_taxonomy(user: User, taxonomy: Taxonomy = None) -> bool:
    """
    Only taxonomy admins can view a taxonomy.
    """
    return is_taxonomy_admin(user)
```

And this test (line 95):

https://github.com/open-craft/openedx-learning/blob/963841c1cf9625b0e6bace6f17e3475e9c1642c4/tests/openedx_tagging/core/tagging/test_rules.py#L82-L96

Sorry for asking for more clarification. Am I misunderstanding you?

Reply:
@rpenido 

> The rule (the one you linked) allows ANY user to view an {enabled=true} taxonomy. That's why I'm confused.

Ah yes, I spoke incorrectly here.. My apologies! Logged-in users *should* be able to list taxonomies, and so they need non-object view perms.

So instead of hacking around this with `has_permission`, let's fix that rule:

```diff
diff --git a/openedx_tagging/core/tagging/rules.py b/openedx_tagging/core/tagging/rules.py
index 052dfdf..4c506a3 100644
--- a/openedx_tagging/core/tagging/rules.py
+++ b/openedx_tagging/core/tagging/rules.py
@@ -19,7 +19,7 @@ def can_view_taxonomy(user: User, taxonomy: Taxonomy = None) -> bool:
     Anyone can view an enabled taxonomy,
     but only taxonomy admins can view a disabled taxonomy.
     """
-    return (taxonomy and taxonomy.enabled) or is_taxonomy_admin(user)
+    return (not taxonomy or taxonomy.enabled) or is_taxonomy_admin(user)


 @rules.predicate
diff --git a/tests/openedx_tagging/core/tagging/test_rules.py b/tests/openedx_tagging/core/tagging/test_rules.py
index 577c594..415ae19 100644
--- a/tests/openedx_tagging/core/tagging/test_rules.py
+++ b/tests/openedx_tagging/core/tagging/test_rules.py
@@ -90,7 +90,7 @@ class TestRulesTagging(TestTagTaxonomyMixin, TestCase):
         assert self.superuser.has_perm("oel_tagging.view_taxonomy", self.taxonomy)
         assert self.staff.has_perm("oel_tagging.view_taxonomy")
         assert self.staff.has_perm("oel_tagging.view_taxonomy", self.taxonomy)
-        assert not self.learner.has_perm("oel_tagging.view_taxonomy")
+        assert self.learner.has_perm("oel_tagging.view_taxonomy")
         assert (
             self.learner.has_perm("oel_tagging.view_taxonomy", self.taxonomy) == enabled
         )
diff --git a/tests/openedx_tagging/core/tagging/test_views.py b/tests/openedx_tagging/core/tagging/test_views.py
index b714015..e282580 100644
--- a/tests/openedx_tagging/core/tagging/test_views.py
+++ b/tests/openedx_tagging/core/tagging/test_views.py
@@ -86,7 +86,7 @@ class TestTaxonomyViewSet(APITestCase):

     @ddt.data(
         (None, status.HTTP_403_FORBIDDEN),
-        ("user", status.HTTP_403_FORBIDDEN),
+        ("user", status.HTTP_200_OK),
         ("staff", status.HTTP_200_OK),
     )
     @ddt.unpack
```